pkg/plugin/vm_backup_item_action.go

```diff
@@ -101,7 +102,21 @@ func (p *VMBackupItemAction) Execute(item runtime.Unstructured, backup *v1.Backu
 	if err != nil {
 		return nil, nil, errors.WithStack(err)
 	}
-	return item, extra, nil
+
+	if vm.Spec.Instancetype != nil && vm.Status.InstancetypeRef != nil && vm.Status.InstancetypeRef.ControllerRevisionRef != nil {
+		vm.Spec.Instancetype.RevisionName = vm.Status.InstancetypeRef.ControllerRevisionRef.Name
```

Comment:
Can you explain why you are updating the vm object?

Reply:
@ShellyKa13 apologies for the delay getting back to this, I didn't think we restored status during restore so this just ensures these values are reflected in the spec itself that should eventually lead to status being recreated correctly.

Reply:
https://velero.io/docs/main/restore-reference/#restore-status-field-of-objects 

> By default, Velero will remove the status field of an object before it’s restored. This is because the value status field is typically set by the controller during reconciliation. However, some custom resources are designed to store environment specific information in the status field, and it is important to preserve such information during restore.

Looks like it's still the default behaviour for Velero unless I've missed where we enable this for VirtualMachines. Happy to drop this if I have!

Reply:
Sure, I think its worth a comment in the code and/or the commit msg

Reply:
https://github.com/kubevirt/kubevirt-velero-plugin/pull/353